internal/provider/vpc_resource.go

```diff
@@ -157,6 +170,41 @@ func (r *vpcResource) Create(ctx context.Context, req resource.CreateRequest, re
 	}
 }
 
+func (r *vpcResource) waitForVPCReadiness(ctx context.Context, id int64, timeouts timeouts.Value) (*tsClient.VPC, error) {
+	tflog.Trace(ctx, "VPCResource.waitForServiceReadiness")
+
+	defaultTimeout := 45 * time.Minute
+	timeout, diags := timeouts.Create(ctx, defaultTimeout)
+	if diags != nil && diags.HasError() {
+		tflog.Error(ctx, fmt.Sprintf("found errs %v", diags.Errors()))
+		return nil, fmt.Errorf("unable to get timeout from config %v", diags.Errors())
+	}
+	conf := retry.StateChangeConf{
+		Pending:                   []string{"CREATING"},
+		Target:                    []string{"CREATED"},
+		Delay:                     5 * time.Second,
+		Timeout:                   timeout,
+		PollInterval:              5 * time.Second,
+		ContinuousTargetOccurence: 1,
+		Refresh: func() (result interface{}, state string, err error) {
+			vpc, err := r.client.GetVPCByID(ctx, id)
+			if err != nil {
+				return nil, "", err
+			}
+			return vpc, vpc.Status, nil
+		},
+	}
+	res, err := conf.WaitForStateContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	vpc, ok := res.(*tsClient.VPC)
+	if !ok {
+		return nil, fmt.Errorf("unexpected type found, expected VPC but got %T", res)
```

Comment:
future improvement: make a dedicated error for unexpected types (this is only the second occurrence so not pressing to do now)

Reply:
true, now im thinking maybe the func could be 

```go
func waitForReadiness[T any](ctx context.Context, id int64, timeouts timeouts.Value, refreshFunc retry.StateRefreshFunc) (*T, error) {
```